src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs

```diff
@@ -44,10 +44,17 @@ public string GetDumpFile()
 
             this.processHelper.WaitForProcessExit(this.procDumpProcess);
 
-            var dumpFile = Path.Combine(this.testResultsDirectory, this.dumpFileName);
-            if (this.fileHelper.Exists(dumpFile))
+            // Dump files can never be more than 1 because procdump will generate single file, but GetFiles function returns an array
+            var dumpFiles = this.fileHelper.GetFiles(this.testResultsDirectory, this.dumpFileName + "*", SearchOption.TopDirectoryOnly);
+            if (dumpFiles.Length > 0)
             {
-                return dumpFile;
+                // Log to diagnostics if multiple files just in case
+                if (dumpFiles.Length != 1)
+                {
+                    EqtTrace.Warning("ProcessDumpUtility.GetDumpFile: Multiple dump files found.");
+                }
+
+                return dumpFiles[0];
```

Comment:
if clean up of dumps did not happen, by returning dumpfiles[0], you could be returning stale dump?